governance/xc-admin/packages/xc-admin-cli/src/index.ts

```diff
@@ -95,4 +101,52 @@ mutlisigCommand(
     await proposeInstructions(squad, vault, [proposalInstruction], false);
   });
 
+mutlisigCommand("upgrade-program", "Upgrade a program from a buffer")
+  .requiredOption(
+    "-p, --program-id <pubkey>",
+    "program that you want to upgrade"
+  )
+  .requiredOption("-b, --buffer <pubkey>", "buffer account")
+
+  .action(async (options: any) => {
+    const wallet = new NodeWallet(
+      Keypair.fromSecretKey(
+        Uint8Array.from(JSON.parse(fs.readFileSync(options.wallet, "ascii")))
+      )
+    );
+    const cluster: PythCluster = options.cluster;
+    const programId: PublicKey = new PublicKey(options.programId);
+    const buffer: PublicKey = new PublicKey(options.buffer);
+    const vault: PublicKey = new PublicKey(options.vault);
+
+    const squad = SquadsMesh.endpoint(getPythClusterApiUrl(cluster), wallet);
+    const msAccount = await squad.getMultisig(vault);
+    const vaultAuthority = squad.getAuthorityPDA(
+      msAccount.publicKey,
+      msAccount.authorityIndex
+    );
+
+    const programDataAccount = PublicKey.findProgramAddressSync(
+      [programId.toBuffer()],
+      BPF_UPGRADABLE_LOADER
+    )[0];
+
+    // This is intruction is not in @solana/web3.js, source : https://docs.rs/solana-program/latest/src/solana_program/bpf_loader_upgradeable.rs.html#200
+    const proposalInstruction: TransactionInstruction = {
+      programId: BPF_UPGRADABLE_LOADER,
+      data: Buffer.from([3, 0, 0, 0]),
```

Comment:
I think we should add a comment explaining why this si the specific byte string we want.

Reply:
Fixed, thanks